sphinx/writers/latex.py

```diff
@@ -396,32 +398,46 @@ def __init__(self, document, builder):
                                     path.basename(builder.config.latex_logo)
         # setup multilingual package
         self.babel = ExtBabel(builder.config.language)
-        if self.elements['polyglossia']:
-            self.elements['babel'] = ''  # disable babel forcely
-            self.elements['multilingual'] = self.elements['polyglossia']
-        elif self.elements['babel']:
-            self.elements['multilingual'] = self.elements['babel']
-            self.elements['classoptions'] += ',' + self.babel.get_language()
-            if builder.config.language:
-                if not self.babel.is_supported_language():
-                    self.builder.warn('no Babel option known for language %r' %
-                                      builder.config.language)
-                self.elements['shorthandoff'] = self.babel.get_shorthandoff()
-                self.elements['fncychap'] = '\\usepackage[Sonny]{fncychap}'
+        if builder.config.language:
+            if not self.babel.is_supported_language():
+                self.builder.warn('no Babel option known for language %r' %
+                                  builder.config.language)
+                self.elements['classoptions'] += ',english'
+                self.elements['multilingual'] = self.elements['babel']
+            else:
+                self.elements['classoptions'] += ',' + self.babel.get_language()
+                if self.elements['polyglossia']:
+                    self.elements['babel'] = ''  # disable babel forcely
+                    self.elements['multilingual'] = self.elements['polyglossia']
+                    self.elements['multilingual'] += ('\n\\setmainlanguage{%s}' %
+                                                      self.babel.get_language())
+                elif self.elements['babel']:
+                    self.elements['multilingual'] = self.elements['babel']
+                    self.elements['shorthandoff'] = self.babel.get_shorthandoff()
+                    # Times fonts don't work with Cyrillic languages
+                    if self.babel.uses_cyrillic():
+                        self.elements['fontpkg'] = ''
 
-                # Times fonts don't work with Cyrillic languages
-                if self.babel.uses_cyrillic():
-                    self.elements['fontpkg'] = ''
+                self.elements['fncychap'] = '\\usepackage[Sonny]{fncychap}'
 
                 # pTeX (Japanese TeX) for support
                 if builder.config.language == 'ja':
                     # use dvipdfmx as default class option in Japanese
                     self.elements['classoptions'] = ',dvipdfmx'
                     # disable babel which has not publishing quality in Japanese
-                    self.elements['babel'] = ''
-                    self.elements['multilingual'] = ''
+                    if self.elements['babel']:
```

Comment:
I added this conditional by error, I will remove it (polyglossia could be ok with Japanese, but anyhow Japanese documents do not use xelatex by default)